cardano-db-sync/src/Cardano/DbSync/Database.hs

```diff
@@ -113,32 +113,35 @@ runActions env actions = do
             else dbAction Continue zs
 
 rollbackLedger :: SyncEnv -> CardanoPoint -> IO (Maybe [CardanoPoint])
-rollbackLedger env point = do
-  mst <- loadLedgerAtPoint (envLedger env) point
-  case mst of
-    Right st -> do
-      let statePoint = headerStatePoint $ headerState $ clsState st
-      -- This is an extra validation that should always succeed.
-      unless (point == statePoint) $
-        logAndPanic (getTrace env) $
-          mconcat
-            [ "Ledger "
-            , textShow statePoint
-            , " and ChainSync "
-            , textShow point
-            , " don't match."
-            ]
-      pure Nothing
-    Left lsfs ->
-      Just . fmap fst <$> verifySnapshotPoint env (OnDisk <$> lsfs)
+rollbackLedger syncEnv point =
+  case envLedgerEnv syncEnv of
+    HasLedger hle -> do
+        mst <- loadLedgerAtPoint hle point
+        case mst of
+          Right st -> do
+            let statePoint = headerStatePoint $ headerState $ clsState st
+            -- This is an extra validation that should always succeed.
+            unless (point == statePoint) $
+              logAndPanic (getTrace syncEnv) $
+                mconcat
+                  [ "Ledger "
+                  , textShow statePoint
+                  , " and ChainSync "
+                  , textShow point
+                  , " don't match."
+                  ]
+            pure Nothing
+          Left lsfs ->
+            Just . fmap fst <$> verifySnapshotPoint syncEnv (OnDisk <$> lsfs)
+    NoLedger _ -> pure Nothing
```

Comment:
Not sure what the scenario should be here of is what I've done is in fact correctby returning `NoLedger _ -> pure Nothing`?

Reply:
I think that's fine. And since the separate cases are handled in this function, the caller place can be siplified, by removing the check
```
            if hasLedgerState env
              then lift $ rollbackLedger env chainSyncPoint
              else pure Nothing
```